lfs/pointer_clean.go

```diff
@@ -31,7 +31,8 @@ func PointerClean(reader io.Reader, fileName string, fileSize int64, cb CopyCall
 	var oid string
 	var size int64
 	var tmp *os.File
-	var exts []*PointerExtension
+	// Initialise to empty set rather than nil for consistency with pointer decode
```

Comment:
Turns out that test is still in my other branch & didn't come across with this. For example:
```
	expected := []*WrappedPointer{
		{Name: "folder/nested.txt", Size: outputs[3].Files[0].Size, Pointer: outputs[3].Files[0]},
		{Name: "folder/nested.txt", Size: outputs[0].Files[2].Size, Pointer: outputs[0].Files[2]},
		{Name: "folder/nested2.txt", Size: outputs[0].Files[3].Size, Pointer: outputs[0].Files[3]},
	}
	// Need to sort to compare equality
	sort.Sort(test.WrappedPointersByOid(expected))
	sort.Sort(test.WrappedPointersByOid(pointers))
	assert.Equal(t, expected, pointers)
```
.. since it's possible to be this specific with the new test setup tools. I don't think I ported the sort utils over to this branch but they're something else you need since the assert lib doesn't have a ConsistsOf() method like Ginkgo/Gomega.

The comment was there to document the reason for changing the line which isn't obvious unless you also look at decodeKV(). An alternative is to change decodeKV() to use nil for extensions by default (that would be my preference actually, avoid unnecessary allocations). I'll take it out of this PR since it's not needed & put it back in as a nil in my upcoming fetch-recent branch.